### PR TITLE
Use a separate communicator for AMAX reduction

### PR DESCRIPTION
**Pull request by erhoo82**
This is to enable the overlap of AMAX reduction and the other DP-communications e.g. gradient reduction and parameter gathering.

### CODE REVIEW

File: apex/transformer/parallel_state.py [hunk 1]
```diff
@@ -84,6 +86,7 @@ def initialize_model_parallel(
     pipeline_model_parallel_size_: int = 1,
     virtual_pipeline_model_parallel_size_: Optional[int] = None,
     pipeline_model_parallel_split_rank_: Optional[int] = None,
+    use_fp8_: Optional[bool] = None,
```
Comment (crcrpar) [inline apex/transformer/parallel_state.py:89]: would the behavior with `use_fp8_=None` be different from that with `use_fp8_=False`?
I'm not sure about the need to make this `Optional` at glance

Comment (crcrpar) [reply apex/transformer/parallel_state.py:89]: If the behavior will be the same, then I'd use 
```suggestion
    use_fp8_: bool = False,
```

File: apex/transformer/parallel_state.py [hunk 2]
```diff
@@ -96,6 +99,7 @@ def initialize_model_parallel(
         pipeline_model_parallel_size: number of GPUs used to parallelize model pipeline.
         virtual_pipeline_model_parallel_size: number of virtual stages (interleaved pipeline).
         pipeline_model_parallel_split_rank: for models with both encoder and decoder, rank in pipeline with split point.
+        use_fp8: FP8 training that needs AMAX reduction across data-parallel ranks.
```
Comment (crcrpar) [inline apex/transformer/parallel_state.py:102]: nit-picking
```suggestion
        use_fp8_: FP8 training that needs AMAX reduction across data-parallel ranks.
```

**Comment by erhoo82** [comment]
Thanks for the comments. I have applied the suggested changes.

**Pull request merged**